packages/scheduled-tasks/src/lib/ScheduledTaskHandler.ts

```diff
@@ -44,10 +44,27 @@ export class ScheduledTaskHandler {
 		);
 	}
 
-	public run(task: string, payload: unknown): Awaitable<unknown> {
+	public async run(task: string, payload: unknown): Promise<unknown> {
 		const piece = this.store.get(task);
 
-		return piece?.run(payload);
+		if (!piece) {
+			container.client.emit(ScheduledTaskEvents.ScheduledTaskNotFound, task, payload);
+			return;
+		}
+
+		const result = await fromAsync(async () => {
+			container.client.emit(ScheduledTaskEvents.ScheduledTaskRun, task, payload);
+			const result = await piece?.run(payload);
```

Comment:
```suggestion
			const result = await piece.run(payload);
```

This is safeguarded by the if check above